bookdown/06-extending-tuner.Rmd

```diff
@@ -0,0 +1,65 @@
+## Adding new Tuners {#extending-tuners}
+
+In this section, we show how to implement a custom tuner for `r mlr_pkg("mlr3tuning")`.
+
+### Adding a new Tuner via bbotk
+
+The `r mlr_pkg("bbotk")` package provides black box optimization algorithms.
+If the new custom tuner is not restricted to hyperparameter tuning, it should be implemented in `r mlr_pkg("bbotk")`.
+The `bbotk::Optimizer` can be easily transformed to a `mlr3tuning::Tuner` later.
+
+1. Check the tuner does not already exist as a `bbotk::Optimizer` or `mlr3tuning::Tuner`.
+2. Use one of the existing optimizer as a [template](#tuner-template).
+3. Add [`.optimize`](#tuner-optimize) private method to optimizer.
+4. Optionally, overwrite the default [`.assign_result`](#tuner-add-result) private method.
+5. Use the [`mlr3tuning::TunerFromOptimizer`](#tuner-from-optimizer) class to transform the `bbotk::Optimizer` to a `mlr3tuning::Tuner`.
+6. Add [unit tests](#tuner-test) for the optimizer and the tuner.
+7. Open a new pull request for the [`bbotk::Optimizer`](https://github.com/mlr-org/bbotk/pulls) and a second one for the [`mlr3tuning::Tuner`](https://github.com/mlr-org/mlr3tuning/pulls).
+
+### Adding a new Tuner via mlr3tuning
+
+If the new custom tuner is only usable for hyperparameter tuning, it should be directly implemented in `r mlr_pkg("mlr3tuning")`.
```

Comment:
What's the alternative to only hyperparameter tuning?

Reply:
I rewrote the introduction. Maybe it is clearer now?